pytest_httpserver/httpserver.py

```diff
@@ -64,6 +64,25 @@ class NoMethodFoundForMatchingHeaderValueError(Error):
     pass
 
 
+class Log(list):
```

Comment:
List inheritance is a very bad idea, if you some of these functionalities you'd rather should use composition

Reply:
Could you be a bit more specific why is it a bad idea?

Reply:
you can google for that question or read about "composition over inheritance"
in few words, your log is not a list, it is not a data structure, it just contains a collection of entries,
so ther should be rather
```
    @property
    def json(self):
        return [json.loads(e.data.decode(obj.charset)) for e in SELF.ENTRIES]
```
exposing list-like interface rather might be a bad design smell, you should expose more domain-specific functions that already operate on this log

Reply:
I can change my PR to make it more open if you want

Reply:
What is a rule of thumb here is that overriding the dunder methods (such as `__getitem__`) is generally a bad idea when inheriting from list, as there are other ways to get the elements of the list  (iterating for example), same applies to `__setitem__` which  can be easily avoided by calling extend method.
In this code, none of the original methods of the list is overridden so it is completely safe to add new methods while keeping the original inherited methods untouched.
When doing composition, it should be made a lot of boilerplate code to make the list to be list-like - I want to provide the list API to the users (I understood that you don't want to use it now) if they want to look at the request objects.
But if I think correctly that your case is solved now, you can use the json property and it will cover your use case. Is that right?